cognite/powerops/transformations/transformations.py

```diff
@@ -203,92 +189,63 @@ def apply(self, time_series_data: pd.Series) -> pd.Series:
         return time_series_data.reindex(union_index).ffill() * non_relative_datapoints
 
 
-class AddWaterInTransit(Transformation):
-    transformation_name = "add_water_in_transit"
-    extra_ts_external_id: str
-    transit_object_name: str  # either plant or gate (not to be confused with object name)
-    _start: Optional[int] = None
-    _end: Optional[int] = None
-    _shape: Optional[Dict[int, float]] = None
-    _discharge: Optional[pd.Series] = None
-
-    @property
-    def start(self):
-        return self._start
-
-    @property
-    def end(self):
-        return self._end
-
-    @property
-    def shape(self):
-        return self._shape
-
-    @property
-    def discharge(self):
-        return self._discharge
+class AddWaterInTransit(DynamicTransformation):
+    discharge_ts_external_id: str
+    transit_object_type: Literal["plant", "gate"]
+    transit_object_name: str
+    shape: Optional[dict[int, float]] = None
+    discharge: Optional[pd.Series] = None
+    start: Optional[int] = None
+    end: Optional[int] = None
```

Comment:
Good use case for pydantic private attributes?
https://docs.pydantic.dev/latest/concepts/models/#private-model-attributes